src/backend/controllers/marketController.js

```diff
@@ -514,3 +564,109 @@ exports.marketTransaction = async (req: Request, res: Response) => {
         }
     }
 }
+
+// Removes a sale from the uses
+exports.removeAMarketSale = async (req: Request, res: Response) => {
+	
+    // Validation
+    const errors = validationResult(req);
+	
+    if (!errors.isEmpty()) {
+        // Return 400 for a bad request
+        return res
+            .status(400)
+            .json({ errors: errors.array() });
+    }
+
+    // Deconstruct request body
+	const { saleId } = req.body;
+
+    // Check if sale exists
+    const sale = await MarketSale.findById(saleId);
+    if (!sale) {
+        console.error('Sale post does not exist');
+        return res
+            .status(400)
+            .json({ msg: 'Sale post does not exist' });
+    }
+
+    // Check if user exists
+    const user = User.findById(sale.sellerId);
+    if (!user) {
+        console.error('User does not exist');
+        return res
+            .status(400)
+            .json({ msg: 'User does not exist' });
+    }
+    
+    // Check between tank and component/casus
+    if (sale.itemType === 'tank') {
+
+        // Check if tank still exists
+        let tank = await Tank.findById(sale.itemId);
+        if (!tank) {
+            console.error('Tank is not in DB');
+            return res
+                .status(404)
+                .json({ msg: 'Tank is not in DB' });
+        }
+
+        // Reassign to seller
+        tank = await Tank.findByIdAndUpdate(sale.itemId, { userId: sale.sellerId }, { new: true });
+        if (!tank) {
+            console.error('Failed to reassign tank');
+            return res
+                .status(500)
+                .json({ msg: 'Failed to reassign tank' });
+        }
+    }
+    // else the post had components or casus blocks
+    else {
+        if (sale.itemType === 'component') {
+            // Give back the amount of components
+            const seller = await User.findById(sale.sellerId);
+
+            seller['inventory']['tankComponents'][sale.itemId] += sale.amount;
```

Comment:
Please don't access object's hardcoded string fields with bracket notation. It is confusing and makes it look like it is an array when it isn't.

`seller.inventory.tankComponents[sale.itemId] += sale.amount` preferred unless there is some reason why this is better.